### ISSUE TITLE
[Build Tools] Update Nodejs Apt Repository to latest

### ISSUE THREAD

**Issue by markmandel** [googleforgames/agones#3430]
Nodejs updated their installation instructions a long time ago, but we didn't update our Dockerfiles (or likely our nodejs versions)!

We use a [deprecated / no longer maintained apt repository](https://github.com/nodesource/distributions/issues/685) for our nodejs installation, and should instead use the newer, [official apt repository](https://github.com/nodesource/distributions#installation-instructions) (and possibly update our nodejs version?).

You can find all the instances here:

https://github.com/search?q=repo%3Agoogleforgames%2Fagones%20%22chris-lea%22&type=code

* [x] https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/build/build-image/Dockerfile#L94
* [x] https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/build/build-sdk-images/node/Dockerfile#L21
* [x] https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/build/build-sdk-images/restapi/Dockerfile#L30 (also not sure why this has nodejs in it - we might want to review this one)
* [x] https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/examples/nodejs-simple/Dockerfile#L20 (will also want to bump the example version, and push the new version)

**Comment by steven-supersolid** [comment]
Oh is this causing the build failure in #3433?

**Comment by steven-supersolid** [comment]
Arg typo in the above comment, LTS is 18, not 10 😱

**Comment by roberthbailey** [comment]
> Oh is this causing the build failure in #3433?

That is the current hypothesis.

**Comment by markmandel** [comment]
Looks like #3434 resolves the top three and unblocks CI!

However, before closing [agones/build/build-sdk-images/restapi/Dockerfile](https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/build/build-sdk-images/restapi/Dockerfile#L30), I'd like to run a test on [agones/build/build-sdk-images/restapi/Dockerfile](https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/build/build-sdk-images/restapi/Dockerfile#L30) where we remove the dependency on nodejs from the image entirely and see if CI passes before closing that checklist item - I have a feeling it's no longer needed for the REST conformance test.

**Comment by markmandel** [comment]
Just untagged the checkbox on [agones/build/build-sdk-images/restapi/Dockerfile](https://github.com/googleforgames/agones/blob/85dd69ab8a3f26a41cfed368a4f843e1473cbb91/build/build-sdk-images/restapi/Dockerfile#L30) - @Kalaiselvi84 can we also do a PR where we remove all of this section:

https://github.com/googleforgames/agones/blob/1caefe08b49ac547ccb1847c7f6784400c159f14/build/build-sdk-images/restapi/Dockerfile#L31-L42

And see what happens? (can just submit as a PR, and let's see if it passes).

I don't see a need anywhere for that nodejs dependency here, so let's give it a shot before closing out that checkbox.

**Comment by markmandel** [comment]
@Kalaiselvi84 just confirming this is all done before closing (pretty sure, but just wanted to double check).

**Comment by Kalaiselvi84** [comment]
One item is pending. I am waiting for your permission to push the nodejs-simple-server image into production.😅

**Comment by markmandel** [comment]
> One item is pending. I am waiting for your permission to push the nodejs-simple-server image into production.😅

![do it](https://media.giphy.com/media/wi8Ez1mwRcKGI/giphy.gif)

**Comment by Kalaiselvi84** [comment]
Done, sir! https://screenshot.googleplex.com/8yanDCtZ5xo6Szp
